asteroid/data/sms_wsj_dataset.py

```diff
@@ -0,0 +1,219 @@
+        nondefault_nsrc (int, optional): Number of sources in the training
+            targets.
+        normalize_audio (bool): If True then both sources and the mixture are
+            normalized with the standard deviation of the mixture.
+    """
+    dataset_name = 'SMS_WSJ'
+
+    def __init__(self, json_path, target, dset, sample_rate=8000,
+                 single_channel=True, segment=4.0, nondefault_nsrc=None,
+                 normalize_audio=False):
+        try:
+            import sms_wsj
+        except:
+            import warnings
+            warnings.warn(
+                "Some of the functionality relies on the sms_wsj package "
+                "downloadable from https://github.com/fgnt/sms_wsj ."
+                "The user is encouraged to install the package"
+            )
+        super().__init__()
+        if target not in SMS_TARGETS.keys():
+            raise ValueError('Unexpected task {}, expected one of '
+                             '{}'.format(target, SMS_TARGETS.keys()))
+
+        # Task setting
+        self.json_path = json_path
+        self.target = target
+        self.target_dict = SMS_TARGETS[target]
+        self.single_channel = single_channel
+        self.sample_rate = sample_rate
+        self.normalize_audio = normalize_audio
+        self.seg_len = None if segment is None else int(segment * sample_rate)
+        if not nondefault_nsrc:
+            self.n_src = self.target_dict['default_nsrc']
+        else:
+            assert nondefault_nsrc >= self.target_dict['default_nsrc']
+            self.n_src = nondefault_nsrc
+        self.like_test = self.seg_len is None
+        self.dset = dset
+
+        # Load json files
+
+        from lazy_dataset.database import JsonDatabase
+        db = JsonDatabase(json_path)
+        dataset = db.get_dataset(dset)
+        # Filter out short utterances only when segment is specified
+        orig_len = len(dataset)
+        drop_utt, drop_len = 0, 0
+        if not self.like_test:
+            def filter_short_examples(example):
+                num_samples = example['num_samples']['observation']
+                if num_samples < self.seg_len:
+                    return False
+                else:
+                    return True
+
+            dataset = dataset.filter(filter_short_examples, lazy=False)
+        self.dataset = dataset
+
+    def __add__(self, sms_wsj):
+        if self.n_src != sms_wsj.n_src:
+            raise ValueError('Only datasets having the same number of sources'
+                             'can be added together. Received '
+                             '{} and {}'.format(self.n_src, sms_wsj.n_src))
+        if self.seg_len != sms_wsj.seg_len:
+            self.seg_len = min(self.seg_len, sms_wsj.seg_len)
+            print('Segment length mismatched between the two Dataset'
+                  'passed one the smallest to the sum.')
+        self.dataset = self.dataset.concatenate(sms_wsj.dataset)
```

Comment:
Genuine question: Do you think this is useful? 
I wrote it for WHAM where you can concatenate clean and noisy separation datasets. But here, the mixtures are always the same and the training targets are different, does it make sense to mix the training targets for one network? Or what other use case would you see?

Reply:
I added this mostly because it was part of the WHAM example and I wanted to add the same amount of customizability.
However, this is basically just one line, so the user can also do this on his own. I will delete it.

Since I am using the lazy_dataset package the user can change everything he wants to change by doing sms_wsj.dataset.concatenate/filter/map etc.